### PR TITLE
Remove old code and metadata

### LINKED ISSUE #383: deps: Remove dependency on sigstore/cosign
**Description**

https://github.com/sigstore/root-signing/pull/380#discussion_r975696399

We currently use some methods from sigstore/cosign (for example, `SignerVerifierFromKeyRef`) and we should ideally only be using methods from sigstore/sigstore.

Also see sigstore-go: https://docs.google.com/document/d/1aZfk1TlzcuaO0uz76M9D26-gAvoZLn0oCAKvkbuhcPM/edit#heading=h.fw3hnltgfsqv

### PR DESCRIPTION
**Pull request by jku**
This is part of #1356: Remove legacy code and metadata. 

It's massive PR, apologies. I can split it if requested but a lot of these parts depend on each other so the split will be mostly aesthetic.

* tuf-on-ci has replaced the go code and bash scripts in this repo
* Makefile is updated to not refer to removed tools
* One missing feature is the key authenticity verification: #1364 
* metadata sources now live in `metadata/`: it's better to remove the legacy `repository/` ASAP so people do not get confused
* I've also removed the ceremony directories: I believe we can let git store the history for those

### CODE REVIEW

**Review by kommendorkapten — APPROVED** [review]
💥

**Comment by jku** [comment]
WRT the go package: Based on GitHub dependency tracking there are no dependents that would break: sigstore-probers is listed as a user but that is AFAICT historical: current sigstore-probers does not use root-signing go code

**Comment by kommendorkapten** [comment]
Good call on the probers, are they in a separate repository?

**Comment by jku** [comment]
> Good call on the probers, are they in a separate repository?

sigstore/sigstore-probers: it used to use some root-signing verify calls but currently the only TUF related thing left should be calls to `theupdateframework/tuf-on-ci/actions/test-repository` action

File: Makefile [hunk 1]
```diff
@@ -18,40 +17,10 @@ help: ## Shows all targets and help from the Makefile (this message).
...
-
-golangci-lint: ## Runs the golangci-lint linter.
-	@set -e;\
-		extraargs=""; \
-		if [ "$(OUTPUT_FORMAT)" == "github" ]; then \
-			extraargs="--out-format github-actions"; \
-		fi; \
-		$(GOBIN)/golangci-lint run -c .golangci.yml ./... $$extraargs
-
 yamllint: ## Runs the yamllint linter.
```
Comment (Hayden-IO) [inline Makefile:20]: In a later PR, probably worth removing the Makefile entirely and moving this block into `.github/workflows/lint.yaml`

**Review by Hayden-IO — APPROVED** [review]
So much red, this is great!

**Pull request merged**